### ISSUE TITLE
Bad rendering and takes ages

### ISSUE THREAD

**Issue by nkh** [henriklovhaug/md-tui#160]
try on https://github.com/nkh/P5-App-Asciio/blob/main/README.md

glow did a good job and loaded instantly while md-tui takes seconds to start displaying anything and it doesn't look good either

**Comment by henriklovhaug** [comment]
Yeah I see the issue. My grammar for markdown does not support these *non-fenced* code blocks. `md-tui` sees them are regular paragraphs and tries to format them. Not sure what the *specification* is there.

It is slow because it tries to fetch all the images, but the links does not point to the image address, but a GitHub page containing the image. Not really much I can do there.

**Comment by nkh** [comment]
Hi, thanks for the fast answer.

Please let me know if the MD doesn't follow specs.

You could render without the images, fetch them in the background, have a short timeout.

**Comment by henriklovhaug** [comment]
There are multiple specs out there. Other readers says they follow Github flavoured markdown for example. Maybe it should be treated as a code block. I honestly don't know.

I could maybe do a fast render without the images and let the fetching happen on a different thread. It sounds like pain to implement, but I could give it shot. 

Anyhow, my current work capacity on this project is severely limited. Final year of uni and all that.

**Comment by Disonantemus** [comment]
> There are multiple specs out there. Other readers says they follow Github flavoured markdown for example. Maybe it should be treated as a code block. I honestly don't know.

&nbsp;

## Spec

1. [Markdown "strict" (**John Gruber**, 2004)](https://daringfireball.net/projects/markdown/basics)

     > To specify an entire block of pre-formatted code, indent every line of the block by 4 spaces or 1 tab.


2. Commonmark: [4.4 Indented code blocks](https://spec.commonmark.org/0.31.2/#indented-code-blocks)

     > An indented code block is composed of one or more indented chunks separated by blank lines. An indented chunk is a sequence of non-blank lines, each preceded by four or more spaces of indentation. The contents of the code block are the literal contents of the lines, including trailing line endings, minus four spaces of indentation. An indented code block has no info string.
     >
     > An indented code block cannot interrupt a paragraph, so there must be a blank line between a paragraph and a following indented code block. (A blank line is not needed, however, between a code block and a following paragraph.)

3. PanDoc's definition: [8.5.1 Indented code blocks](https://pandoc.org/chunkedhtml-demo/8.5-verbatim-code-blocks.html#indented-code-blocks)

     > A block of text indented four spaces (or one tab) is treated as verbatim text: that is, special characters do not trigger special formatting, and all spaces and line breaks are preserved. For example,
     > 
     >     if (a > 3) {
     >       moveShip(5 * gravity, DOWN);
     >     }
     > 
     > The initial (four space or one tab) indentation is not considered part of the verbatim text, and is removed in the output.
     > 
     > Note: blank lines in the verbatim text need not begin with four spaces.

---

## Output from similar terminal apps

Excerpt from **README.md**

~~~
sed -n '9,15p' README.md
# SYNOPSIS

         ___                     
        /   |  __________________
       / /| | / ___/ ___/ / / __ \
      / ___ |(__  ) /__/ / / /_/ /
     /_/  |_/____/\___/_/_/\____/
~~~

Rendered by **md-tui**

~~~
                                              SYNOPSIS

           ___

          /   |  __________________        / /| | / ___/ ___/ / / __ \       / ___ |(__  ) /__/ / /
  /_/ /      /_/  |_/____/\___/_/_/\____/
~~~

Rendered by **glow**

~~~
$ sed -n '9,15p' README.md | glow

   SYNOPSIS

         ___
        /   |  __________________
       / /| | / ___/ ___/ / / __ \
      / ___ |(__  ) /__/ / / /_/ /
     /_/  |_/____/\___/_/_/\____/
~~~

Rendered by **lowdown**

~~~
$ sed -n '9,15p' README.md | lowdown -stterm
    SYNOPSIS

      │      ___
      │     /   |  __________________
      │    / /| | / ___/ ___/ / / __ \
      │   / ___ |(__  ) /__/ / / /_/ /
      │  /_/  |_/____/\___/_/_/\____/
~~~

Rendered by **mdcat**

~~~
$ sed -n '9,15p' README.md | mdcat
┄SYNOPSIS

────────────────────
     ___
    /   |  __________________
   / /| | / ___/ ___/ / / __ \
  / ___ |(__  ) /__/ / / /_/ /
 /_/  |_/____/\___/_/_/\____/
────────────────────
~~~

Rendered by **pandoc**

~~~
$ sed -n '9,15p' README.md | pandoc -t plain
SYNOPSIS

         ___
        /   |  __________________
       / /| | / ___/ ___/ / / __ \
      / ___ |(__  ) /__/ / / /_/ /
     /_/  |_/____/\___/_/_/\____/
~~~

Rendered by **mdv**

~~~
$ sed -n '9,15p' README.md | mdv -

SYNOPSIS

  |  ___
  |      /   |  __________________
  |     / /| | / ___/ ___/ / / __ \
  |    / ___ |(__  ) /__/ / / /_/ /
  |   /_/  |_/____/\___/_/_/\____/
~~~

Rendered by **mdformat** (is a formatter, convert "indented-code-blocks" to "fenced-code-block")

~~~
$ sed -n '9,15p' README.md | mdformat -
# SYNOPSIS

```
     ___
    /   |  __________________
   / /| | / ___/ ___/ / / __ \
  / ___ |(__  ) /__/ / / /_/ /
 /_/  |_/____/\___/_/_/\____/
```
~~~

---

## Online

- [Markdown (strict)](https://daringfireball.net/projects/markdown/dingus)
- [Commonmark](https://spec.commonmark.org/dingus/)
- [Github](https://github.com/)
- [Dilinger](https://dillinger.io/)
- [StackEdit](https://stackedit.io/)
- Many (many) more ...
